server/app/controllers/applicant/RedirectController.java

```diff
@@ -69,37 +72,71 @@ public CompletionStage<Result> programByName(Http.Request request, String progra
       return CompletableFuture.completedFuture(result);
     }
 
-    CompletableFuture<Applicant> applicantFuture = profile.get().getApplicant();
-    CompletableFuture<Program> programFuture = programRepository.getForSlug(programName);
-
-    return CompletableFuture.allOf(applicantFuture, programFuture)
-        .thenApplyAsync(
-            empty -> {
-              if (applicantFuture.isCompletedExceptionally()) {
-                return notFound();
-              } else if (programFuture.isCompletedExceptionally()) {
-                return notFound();
-              }
-
-              Applicant applicant = applicantFuture.join();
+    return profile
+        .get()
+        .getApplicant()
+        .thenComposeAsync(
+            (Applicant applicant) -> {
               // Attempt to set default language for the applicant.
               applicant = languageUtils.maybeSetDefaultLocale(applicant);
+              final long applicantId = applicant.id;
```

Comment:
Not blocking: Any particular reason to stash this off here rather than just accessing it on the Applicant object at the callsite?

Reply:
variables defined in an outer scope and referenced in lambdas must be final, applicant can't be final because we reassign it on the line above